JavaScript/JavaScriptSDK.Tests/CheckinTests/SendBuffer.tests.ts

```diff
@@ -0,0 +1,420 @@
+﻿/// <reference path="..\TestFramework\Common.ts" />
+/// <reference path="../../JavaScriptSDK/sender.ts" />
+/// <reference path="../../javascriptsdk/appinsights.ts" />
+
+class SendBufferTests extends TestClass {
+
+    private getArraySendBuffer: (emitLineDelimitedJson?: boolean) => Microsoft.ApplicationInsights.ArraySendBuffer;
+    private getSessionStorageSendBuffer: (emitLineDelimitedJson?: boolean) => Microsoft.ApplicationInsights.SessionStorageSendBuffer;
+
+    public testCleanup() {
+        Microsoft.ApplicationInsights.DataLossAnalyzer.enabled = false;
+    }
+
+    public testInitialize() {
+        if (Microsoft.ApplicationInsights.Util.canUseSessionStorage()) {
+            sessionStorage.clear();
+        }
+
+        var config: Microsoft.ApplicationInsights.ISenderConfig = {
+            emitLineDelimitedJson: () => false,
+            storeSendBufferInSessionStorage: () => false,
+            endpointUrl: () => null,
+            maxBatchSizeInBytes: () => null,
+            maxBatchInterval: () => null,
+            disableTelemetry: () => null
+        };
+
+        this.getArraySendBuffer = (emitLineDelimitedJson?: boolean) => {
+            if (emitLineDelimitedJson) {
+                config.emitLineDelimitedJson = () => emitLineDelimitedJson;
```

Comment:
I'm just curious but why are functions being used for simple values, are there instances where there is more logic applied to these kinds of "properties"?